deployer/cloudflaredeployer.go

```diff
@@ -0,0 +1,281 @@
+		if err := createRoute(api, routerRoute, routerScriptName, job.domain); err != nil {
+			log.LogWarn(fmt.Sprintf("error while creating route: %s", err))
+		}
+
+		log.LogDone(fmt.Sprintf("deployed to Cloudflare at %s", routerRoute))
+	}
+
+	return nil
+}
+
+func getAPIClient() (*cloudflare.API, error) {
+	apiToken := os.Getenv("CF_API_TOKEN")
+	accountId := os.Getenv("CF_ACCOUNT_ID")
+
+	if apiToken == "" || accountId == "" {
+		return nil, errors.New("missing CF_API_TOKEN or CF_ACCOUNT_ID")
+	}
+
+	api, err := cloudflare.NewWithAPIToken(apiToken, cloudflare.UsingAccount(accountId))
+	if err != nil {
+		return nil, err
+	}
+
+	return api, nil
+}
+
+func getRunnablesByName(ctx *project.Context, meta appsource.Meta) map[string]directive.Runnable {
+	runnablesByName := make(map[string]directive.Runnable)
+	runnables := ctx.AppSource.Runnables(meta.Identifier, meta.AppVersion)
+
+	for _, runnable := range runnables {
+		runnablesByName[runnable.Name] = runnable
+	}
+
+	return runnablesByName
+}
+
+func renderRouterWorker(workersToRoute map[string]cfWorker) (string, error) {
+	t, err := template.New("router_worker").Parse(routerWorkerTmpl)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse template")
+	}
+
+	type routerWorkerData struct {
+		IttyRouter     string
+		WorkersToRoute map[string]cfWorker
+	}
+
+	data := routerWorkerData{
+		IttyRouter:     ittyRouter,
+		WorkersToRoute: workersToRoute,
+	}
+
+	var res bytes.Buffer
+
+	if err := t.Execute(&res, data); err != nil {
+		return "", errors.Wrap(err, "failed to render template")
+	}
+
+	return res.String(), nil
+}
+
+func renderRequestWorker(method string, path string) (string, error) {
+	t, err := template.New("request_worker").Parse(requestWorkerTmpl)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse template")
+	}
+
+	type requestWorkerData struct {
+		Method     string
+		Path       string
+		IttyRouter string
+		Shared     string
+	}
+
+	data := requestWorkerData{
+		Method:     strings.ToLower(method),
+		Path:       path,
+		IttyRouter: ittyRouter,
+		Shared:     shared,
+	}
+
+	var res bytes.Buffer
+
+	if err := t.Execute(&res, data); err != nil {
+		return "", errors.Wrap(err, "failed to render template")
+	}
+
+	return res.String(), nil
+}
+
+func deployRunnable(api *cloudflare.API, runnable *directive.Runnable, _type string, script string, zoneName string) (string, error) {
+	zoneID, err := api.ZoneIDByName(zoneName)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get zone id")
+	}
+
+	data, err := runnable.ModuleRef.Bytes()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get module ref")
+	}
+	wasmReader := bytes.NewReader(data)
+
+	scriptParams := cloudflare.WorkerScriptParams{
+		Script: script,
+		Bindings: map[string]cloudflare.WorkerBinding{
+			"WASM_MODULE": cloudflare.WorkerWebAssemblyBinding{Module: wasmReader},
+		},
+	}
+
+	scriptName := fmt.Sprintf("subo-%s-%s", _type, runnable.Name)
```

Comment:
This should probably be based on the `ApplicationIdentifier` instead of `subo-` so that it's unique?

Reply:
done